interfaces/MatlabToolbox/matlab_new/Thermo/ThermoPhase.m

```diff
@@ -0,0 +1,1736 @@
+classdef ThermoPhase < handle
+    
+    properties
+        tpID
+        T % temperature
+        P % pressure
+        D % density
+        X % mole fractions
+        Y % mass fractions
+        H % enthalpy
+        S % entropy
+        U % internal energy
+        G % Gibbs free energye
+        basis
+    end
+
+    properties (Dependent)
+        V % specific volume
+        DP
+        DPX
+        DPY
+        HP
+        HPX
+        HPY
+        PV
+        PVX
+        PVY
+        SH
+        SHX
+        SHY
+        SP
+        SPX
+        SPY
+        ST
+        STX
+        STY
+        SV
+        SVX
+        SVY
+        TD
+        TDX
+        TDY
+        TH
+        THX
+        THY
+        TP
+        TPX
+        TPY
+        TV
+        TVX
+        TVY
+        UV
+        UVX
+        UVY
+        UP
+        UPX
+        UPY
+        VH
+        VHX
+        VHY
+    end
+    
+    methods
+        %% ThermoPhase class constructor
+
+        function tp = ThermoPhase(src, id)
+            % THERMOPHASE  ThermoPhase class constructor.
+            % t = ThermoPhase(src, id)
+            %
+            % :param src:
+            %     Input string of YAML, CTI, or XML file name.
+            % :param id:
+            %     ID of the phase to import as specified in the input file. (optional)
+            % :return:
+            %     Instance of class :mat:func:`ThermoPhase`
+            %
+            checklib;
+            if nargin > 2
+                error('ThermoPhase expects 1 or 2 input arguments.');
+            end
+            if nargin == 1
+                id = '-';
+            end
+            tp.tpID = calllib(ct, 'thermo_newFromFile', src, id);
+            tp.basis = 'molar';
+        end
+        
+        %% Utility methods
+
+        function display(tp, threshold)
+            % Display thermo properties
+            
+            if nargin < 2 || ~isnumeric(threshold)
+                threshold = 1e-14;
+            end
+            calllib(ct, 'thermo_print', tp.tpID, 1, threshold);
+        end
```

Comment:
Given that you don't have a way to set a callback to a Matlab function via the `clib` interface, you need to use the `thermo_report` here instead, which will return the report as a string that you can then print in Matlab. The reason you are seeing inconsistent results here is because the meaning of "stdout" is different on different systems, and only sometimes will correspond to the same location where the rest of Matlab's text output is going. Unfortunately, I think this means that there is no way to reliably get logger output from Cantera into Matlab while using the `clib` interface.